### PR TITLE
Adds a muzzle-breathmask

### PR DESCRIPTION
**Pull request by Greniza**
## About The Pull Request

This PR adds a muzzle-breathmask, and has it replace the standard medical mask present in the anesthetic closet.

## Why It's Good For The Game

Provides a way of ~~dragging kidnap victims through space without them suffocating or screaming~~ performing surgery using a mask which silences their grunts of pain before they go under.

It also attaches the muzzle code which previously blocked speech to a clothing flag rather than the muzzle item.

## Changelog
:cl:
add: added non-adjustable mask which allows internals, is sterile, and muzzles a patient or victim, complete with sprites
imageadd: replaced standard sterile breathmask in anesthetic closet with above muzzle breathmask
add: adds BLOCKS_SPEECH clothing flag and reworked muzzle code and is_muzzled() function be based on flag
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
You're adding a straight-upgrade item to round start. Tradeoffs are generally good design.
IMO if you really want to have this in-game for whatever reason, add it to RND so people actually have to work for better items.

**Comment by CRITAWAKETS** [comment]
literally just fetish content, coulda just added in cockvore

**Comment by Greniza** [comment]
> You're adding a straight-upgrade item to round start. Tradeoffs are generally good design.
> IMO if you really want to have this in-game for whatever reason, add it to RND so people actually have to work for better items.

@MrDoomBringer
I mean, it's currently _exclusively_ accessible through an anesthetic cabinet (under normal circumstances) at round start, so it's not like it's a common item. It's also got a pretty niche use case, as do muzzles. It has a worse force equip time than muzzles (25 as opposed to 20); in many scenarios, this makes it strictly worse than a muzzle. It seems balanced enough to me thanks to its rarity and marginally worse stats.

**Comment by MrDoomBringer** [comment]
it taking 0.5 seconds longer to equip means pretty much nothing given that people only ever get muzzled when they're already cuffed
also, having it be exclusive to the locker doesn't affect anything as (admittedly, from my experience) people barely use muzzles anyways - so there won't really be that much demand for them in the first place

tldr this is a super minor addition but still doesn't seem to be that well thought out

**Comment by Serxule** [comment]
> literally just fetish content, coulda just added in cockvore

i have suffocated due to sec not giving me internals many times, including a few times where they muzzled me, I wouldn't say its fetish content

**Comment by Greniza** [comment]
> it taking 0.5 seconds longer to equip means pretty much nothing given that people only ever get muzzled when they're already cuffed
> also, having it be exclusive to the locker doesn't affect anything as (admittedly, from my experience) people barely use muzzles anyways - so there won't really be that much demand for them in the first place
> 
> tldr this is a super minor addition but still doesn't seem to be that well thought out

Another thing of note is that the medical department starts with medical breath masks, which are sterile. They are a strict upgrade from a run-of-the-mill breath mask. Most roles have direct upgrades at roundstart from what is the 'default'; and, as you said, muzzles are very rarely used. It's a marginally better item than a muzzle, so it's marginally harder to obtain. I would think that would balance out.

You say that there's no demand for this item. That argument makes sense in the context of telling someone to not write code for that item. However, this code has already been written, sprites already painted. At this point, why not add it?

Conversely to what you think, making this into an R&D item would make it easier to obtain. This isn't a major enough item to justify putting it under a high-cost tech, so making it an R&D item would make it printable. Then, a chemist (for example) would be able to obtain it without illicit access, which they could not do before.

**Comment by GuillaumePrata** [comment]
This does allow you to muzzle plasmaman without them suffocating if you want to do that and go around the helmet removing issue...

File: code/modules/clothing/masks/miscellaneous.dm [hunk 1]
```diff
@@ -16,6 +16,16 @@
 			return
 	..()
 
+/obj/item/clothing/mask/muzzle/breath // Must be under /mask/muzzle/ because the speech obstruction is tied to /mask/muzzle rather than being a cloting flag. *sigh
```
Comment (Mothblocks) [inline code/modules/clothing/masks/miscellaneous.dm:19]: :-1:

Comment (Mothblocks) [reply code/modules/clothing/masks/miscellaneous.dm:19]: Make it an element.

**Comment by Fikou** [comment]
> > literally just fetish content, coulda just added in cockvore
> 
> i have suffocated due to sec not giving me internals many times, including a few times where they muzzled me, I wouldn't say its fetish content

when will sec ever go to medbay and open the anaesthetic closet they dont have access to

File: code/modules/clothing/masks/miscellaneous.dm (hunk 1 above)
Comment (Greniza) [reply code/modules/clothing/masks/miscellaneous.dm:19]: Could I add it as a clothing flag, and then alter the /mob/living/carbon/is_muzzled() function to check for this clothing flag?

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -208,7 +208,11 @@
 	loc.handle_fall(src)//it's loc so it doesn't call the mob's handle_fall which does nothing
 
 /mob/living/carbon/is_muzzled()
-	return(istype(src.wear_mask, /obj/item/clothing/mask/muzzle))
+	/// if this is bad runtime, then I'm sorry. I don't know how to make it both universal and efficient.
```
Comment (Greniza) [inline code/modules/mob/living/carbon/carbon.dm:211]: My logic, here, is that you could theoretically imagine an item on most clothing categories (except for, like, glasses and boots) which blocks speech. I want to be able to account for those eventualities, in case someone else wants to create an item which blocks speech.

File: code/_globalvars/bitfields.dm [hunk 3]
```diff
@@ -89,6 +89,7 @@ DEFINE_BITFIELD(clothing_flags, list(
 	"THICKMATERIAL" = THICKMATERIAL,
 	"VOICEBOX_DISABLED" = VOICEBOX_DISABLED,
 	"VOICEBOX_TOGGLABLE" = VOICEBOX_TOGGLABLE,
+	"BLOCKS_SPEECH" = BLOCKS_SPEECH,
```
Comment (Mothblocks) [inline code/_globalvars/bitfields.dm:92]: Alphabetical order.

File: code/modules/clothing/masks/miscellaneous.dm [hunk 4]
```diff
@@ -3,6 +3,7 @@
 	desc = "To stop that awful noise."
 	icon_state = "muzzle"
 	inhand_icon_state = "blindfold"
+	clothing_flags = BLOCKS_SPEECH // I'd recommend a double-check on this flag before it goes live.
```
Comment (Mothblocks) [inline code/modules/clothing/masks/miscellaneous.dm:6]: This comment isn't necessary, what you put in code should be what is able to go into production--comments like this should be best left in a comment on the PR.

Comment (Mothblocks) [reply code/modules/clothing/masks/miscellaneous.dm:6]:
```suggestion
	clothing_flags = BLOCKS_SPEECH
```

File: code/modules/clothing/masks/miscellaneous.dm [hunk 5]
```diff
@@ -16,6 +17,25 @@
...
+	desc = "To silence those pesky patients before putting them under."
+	icon_state = "breathmuzzle"
+	inhand_icon_state = "breathmuzzle"
+	clothing_flags = MASKINTERNALS | BLOCKS_SPEECH
+	gas_transfer_coefficient = 0.1
+	permeability_coefficient = 0.01
+	equip_delay_other = 25 // my sprite has 4 straps, a-la a head harness. takes a while to equip, longer than a muzzle
+
+/obj/item/clothing/mask/breathmuzzle/attack_paw(mob/user, list/modifiers)
+	/// The breathmuzzle is similar enough to a regular muzzle that similar rules would apply to both.
```
Comment (Mothblocks) [inline code/modules/clothing/masks/miscellaneous.dm:31]:
```suggestion
	// The breathmuzzle is similar enough to a regular muzzle that similar rules would apply to both.
```

`///` is for doc comments.

File: code/modules/clothing/masks/miscellaneous.dm [hunk 6]
```diff
@@ -16,6 +17,25 @@
...
+	if(iscarbon(user))
+		var/mob/living/carbon/carbon_user = user
+		if(src == carbon_user.wear_mask)
+			to_chat(user, "<span class='warning'>You need help taking this off!</span>")
```
Comment (Mothblocks) [inline code/modules/clothing/masks/miscellaneous.dm:35]: Really, this should be an element as well, but only having it in these two places is fine enough for now.

File: code/modules/mob/living/carbon/carbon.dm (hunk 2 above)
Comment (Mothblocks) [reply code/modules/mob/living/carbon/carbon.dm:211]: It's fine, but removing `is_muzzled` and creating an element for this is much better. I won't require it now, but it's something we should do eventually. Remove the comment.

File: code/__DEFINES/obj_flags.dm [hunk 7]
```diff
@@ -54,6 +54,7 @@
 #define ANTI_TINFOIL_MANEUVER   (1<<12) //Hats with negative effects when worn (i.e the tinfoil hat).
 #define DANGEROUS_OBJECT    (1<<13)     //Clothes that cause a larger notification when placed on a person.
 #define LARGE_WORN_ICON    (1<<14)     //Clothes that use large icons, for applying the proper overlays like blood
+#define BLOCKS_SPEECH (1<<15) // Clothes that block speech (i.e the muzzle). Can be applied to any clothing piece.
```
Comment (Mothblocks) [inline code/__DEFINES/obj_flags.dm:57]:
```suggestion
/// Clothes that block speech (i.e the muzzle). Can be applied to any clothing piece.
#define BLOCKS_SPEECH (1<<15)
```

Doc comment.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/__DEFINES/obj_flags.dm (hunk 7 above)
Comment (Greniza) [reply code/__DEFINES/obj_flags.dm:57]: If you say so, it'd break the standard format of the document.

**Comment by Greniza** [comment]
It says that Mothblocks requested changes- I have completed those changes through modifying the files (rather than going through the thing on GitHub, that would've saved a minute or two. Oops.).

File: code/__DEFINES/obj_flags.dm (hunk 7 above)
Comment (Mothblocks) [reply code/__DEFINES/obj_flags.dm]: The rest of the document should be changed at some point, like I did to traits.

**Comment by Mothblocks** [comment]
Test failure isn't relevant here.
